### PR TITLE
EDIFNetlist.{copy,migrate}CellAndSubCells() to copy into new same-named library

### LINKED ISSUE #512: EDIFNetlist.copyCellAndSubCells() does not copy library correctly
EDIFNetlist.copyCellAndSubCells fails to copy some libraries associated with the given cell.

### PR DESCRIPTION
**Pull request by eddieh-xlnx**
Rather than into the work library, if source library doesn't exist in destination netlist.

Fixes #512.

### CODE REVIEW

**Comment by eddieh-xlnx** [comment]
Should `migrateCellAndSubCells()` get the same treatment?

File: src/com/xilinx/rapidwright/edif/EDIFNetlist.java [hunk 1]
```diff
@@ -553,7 +553,7 @@ private EDIFCell copyCellAndSubCellsWorker(EDIFCell cell, Set<EDIFCell> copiedCe
 			if(cell.getLibrary().isHDIPrimitivesLibrary()){
 				destLib = getHDIPrimitivesLibrary();
 			}else{
-				destLib = getWorkLibrary();
+				destLib = addLibrary(new EDIFLibrary(cell.getLibrary().getName()));
```
Comment (clavin-xlnx) [inline src/com/xilinx/rapidwright/edif/EDIFNetlist.java:556]: There might need to be a check here to see if the destination netlist already has the same named library.  If it does, we should just add the cell to it.

**Comment by clavin-xlnx** [comment]
> Should `migrateCellAndSubCells()` get the same treatment?

Yes, that would probably be wise.

File: src/com/xilinx/rapidwright/edif/EDIFNetlist.java (hunk 1 above)
Comment (eddieh-xlnx) [reply src/com/xilinx/rapidwright/edif/EDIFNetlist.java:556]: That does get checked here a few lines up:

https://github.com/Xilinx/RapidWright/blob/c2770a77b7481745abc246b3f6cad61d722bc70f/src/com/xilinx/rapidwright/edif/EDIFNetlist.java#L551-L552

Comment (clavin-xlnx) [reply src/com/xilinx/rapidwright/edif/EDIFNetlist.java:556]: Sorry, I missed that.  Thanks!

**Review by clavin-xlnx — APPROVED** [review]

**Pull request merged**